src/TableFunctions/TableFunctionAzureBlobStorage.cpp

```diff
@@ -59,123 +57,119 @@ StorageAzureBlob::Configuration TableFunctionAzureBlobStorage::parseArgumentsImp
-        if (is_format_arg(fourth_arg))
+        else if (engine_args.size() == 5)
         {
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Format, compression and structure must be last arguments");
+            auto fourth_arg = checkAndGetLiteralArgument<String>(engine_args[3], "format/account_name");
+            if (is_format_arg(fourth_arg))
+            {
+                configuration.format = fourth_arg;
+                configuration.compression_method = checkAndGetLiteralArgument<String>(engine_args[4], "compression");
+            }
+            else
+            {
+                configuration.account_name = fourth_arg;
+                configuration.account_key = checkAndGetLiteralArgument<String>(engine_args[4], "account_key");
+            }
         }
-        else
+        else if (engine_args.size() == 6)
         {
-            configuration.account_name = fourth_arg;
-            configuration.account_key = checkAndGetLiteralArgument<String>(engine_args[4], "account_key");
-            auto sixth_arg = checkAndGetLiteralArgument<String>(engine_args[5], "format/account_name");
-            if (!is_format_arg(sixth_arg))
-                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unknown format {}", sixth_arg);
-            configuration.format = sixth_arg;
-            configuration.compression_method = checkAndGetLiteralArgument<String>(engine_args[6], "compression");
+            auto fourth_arg = checkAndGetLiteralArgument<String>(engine_args[3], "format/account_name");
+            if (is_format_arg(fourth_arg))
+            {
+                configuration.format = fourth_arg;
+                configuration.compression_method = checkAndGetLiteralArgument<String>(engine_args[4], "compression");
+                configuration.structure = checkAndGetLiteralArgument<String>(engine_args[5], "structure");
+            }
+            else
+            {
+                configuration.account_name = fourth_arg;
+                configuration.account_key = checkAndGetLiteralArgument<String>(engine_args[4], "account_key");
+                auto sixth_arg = checkAndGetLiteralArgument<String>(engine_args[5], "format/account_name");
+                if (!is_format_arg(sixth_arg))
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unknown format {}", sixth_arg);
+                configuration.format = sixth_arg;
+            }
         }
-    }
-    else if (engine_args.size() == 8)
-    {
-
-        auto fourth_arg = checkAndGetLiteralArgument<String>(engine_args[3], "format/account_name");
-        if (is_format_arg(fourth_arg))
+        else if (engine_args.size() == 7)
         {
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Format and compression must be last arguments");
+            auto fourth_arg = checkAndGetLiteralArgument<String>(engine_args[3], "format/account_name");
+            if (is_format_arg(fourth_arg))
```

Comment:
I think it's redundunt. When we have 7 arguments, there is only one possible signature
`azureBlobStorage(connection_string|storage_account_url, container_name, blobpath, account_name, account_key, format, compression`, we don't need to check if `fourth_arg` is format or not

Reply:
addressed it